### PR TITLE
I/O names must be unique across multiple conversions

### PR DESCRIPTION
**Pull request by peterschrammel**
Necessary for incremental unwinding, which performs incremental conversions of symex_target_equationt.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-symex/symex_target_equation.h [hunk 1]
```diff
@@ -405,6 +405,9 @@ class symex_target_equationt:public symex_targett
   // for enforcing sharing in the expressions stored
   merge_irept merge_irep;
   void merge_ireps(SSA_stept &SSA_step);
+
+  // for unique I/O names
+  std::size_t io_count = 0;
```
Comment (tautschnig) [inline src/goto-symex/symex_target_equation.h:410]: Ok, but possibly `symex_target_equationt` really never was the right place to do this? To me, this looks just like what we're doing with renaming nondets, and I am thus wondering whether `goto_symext` would really be the right place to take care of this?

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 1b57b00fcf2b37e0b10c5ac5cb5cb5037336a597).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99559432

**Comment by kroening** [comment]
I am somewhat worried; this will be a very surprising side effect.

**Comment by peterschrammel** [comment]
Same for function call arguments...

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: af514dc3ac897d7268e6840bd77c00bd2c40d960).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99569699

**Comment by peterschrammel** [comment]
> I am somewhat worried; this will be a very surprising side effect.

@kroening, what does your comment refer to?

**Pull request merged**